### PR TITLE
Add GTMTimeUtils

### PR DESCRIPTION
**Pull request by dmaclach**
Utilities for relatively common desire to know the launch time of an app, or the boot time of a device.

### CODE REVIEW

**Comment by dmaclach** [comment]
Looks like a Travis buildbot failure?

File: Foundation/GTMTimeUtils.m [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+NSTimeInterval GTMTimeValToNSTimeInterval(struct timeval time) {
+  return time.tv_sec + (time.tv_usec / (double)USEC_PER_SEC);
+}
+
+struct timeval GTMBootTimeRelativeTo1970(void) {
+  struct timeval bootTime = { 0, 0 };
+  int mib[2] = { CTL_KERN, KERN_BOOTTIME };
+  size_t bootTimeSize = sizeof(bootTime);
+  if (sysctl(mib, 2, &bootTime, &bootTimeSize, NULL, 0) != 0) {
+    NSCAssert(errno == 0, @"sysctl error - %d", errno);
```
Comment (thomasvl) [inline Foundation/GTMTimeUtils.m]: These will be at the mercy of the developers setting to compile out in release?  Do you want to use our DevAssert to ensure they default to DEBUG only?

File: Foundation/GTMTimeUtils.m [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    struct timeval invalid = { 0, 0 };
+    return invalid;
+  }
+  return info.kp_proc.p_starttime;
+}
+
+NSDate *GTMAppLaunchDate() {
+  NSTimeInterval ti =
+  GTMTimeValToNSTimeInterval(GTMAppLaunchTimeRelativeTo1970());
```
Comment (thomasvl) [inline Foundation/GTMTimeUtils.m]: fix indent

File: Foundation/GTMTimeUtils.m [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  return [NSDate dateWithTimeIntervalSince1970:ti];
+}
+
+NSDate *GTMBootDate() {
+  NSTimeInterval ti =
+  GTMTimeValToNSTimeInterval(GTMBootTimeRelativeTo1970());
```
Comment (thomasvl) [inline Foundation/GTMTimeUtils.m]: Fix indent

File: Foundation/GTMTimeUtils.m (hunk 1 above)
Comment (dmaclach) [reply Foundation/GTMTimeUtils.m]: Done

File: Foundation/GTMTimeUtils.m (hunk 2 above)
Comment (dmaclach) [reply Foundation/GTMTimeUtils.m]: Done

File: Foundation/GTMTimeUtils.m (hunk 3 above)
Comment (dmaclach) [reply Foundation/GTMTimeUtils.m]: Done

**Pull request merged**